### PR TITLE
Feature 2 object model

### PR DESCRIPTION
**Pull request by KacFlor**

### CODE REVIEW

File: src/main/java/com/example/demo/CardData/CardData.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.example.demo.CardData;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/CardData/CardData.java]: lets change project name to something different, this is not example demo

File: src/main/java/com/example/demo/Category/Category.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Setter;
+
+@Entity
+@Table
+@Getter
+@Setter
+public class Category {
+
+    private Integer id;
+    private  String name;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Category/Category.java]: Extra space, install some kind of plugin to intellij to auto format java code

File: src/main/java/com/example/demo/DemoApplication.java [hunk 3]
```diff
@@ -1,5 +1,6 @@
 package com.example.demo;
 
+import com.example.demo.User.User;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/DemoApplication.java]: ?

File: src/main/java/com/example/demo/CardData/CardDataService.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+@Service
+public class CardDataService {
+
+    @Autowired
+    public CardDataService() {
+
+    }
+
+    public List<CardData> getCardData() {return List.of();}
```
Comment (Ficzer) [inline src/main/java/com/example/demo/CardData/CardDataService.java]: Shouldn't we get data from reporsitory? List.of() just return empty list right?

File: src/main/java/com/example/demo/student/Student.java [hunk 5]
```diff
@@ -7,6 +7,7 @@
 @Entity
```
Comment (Ficzer) [inline src/main/java/com/example/demo/student/Student.java]: Why do we need it? It was an example

File: src/main/java/com/example/demo/CardData/CardData.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table
+@Getter
+@Setter
+public class CardData {
+
+    private Integer id;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/CardData/CardData.java]: Doesn't it need to be specified as a table id by some kind of annotation?

File: src/main/java/com/example/demo/Order_item/Order_item.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package com.example.demo.Order_item;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Order_item/Order_item.java]: ??? duplicate

Comment (Ficzer) [reply src/main/java/com/example/demo/Order_item/Order_item.java]: Remove this class

File: src/main/java/com/example/demo/Product/Product.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Setter;
+
+@Entity
+@Table
+@Getter
+@Setter
+public class Product {
+
+    private Integer id;
+    private String SKU;
```
Comment (Ficzer) [inline src/main/java/com/example/demo/Product/Product.java]: use lower case

File: pom.xml [hunk 9]
```diff
@@ -1,12 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (Ficzer) [inline pom.xml:1]: Typical project structure is that we put all object model classes in one dir, services in another dir, repositories in another itp. Lets also do some refactor and create base entity class, every object model can extend it, see https://github.com/spring-projects/spring-petclinic/tree/main/src/main/java/org/springframework/samples/petclinic/model

File: src/main/java/com/example/demo/Order_item/Order_item.java (hunk 7 above)
Comment (Ficzer) [reply src/main/java/com/example/demo/Order_item/Order_item.java]: Next time before you send me a MR check what you have here please

File: src/main/java/com/example/demo/student/Student.java (hunk 5 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/student/Student.java]: done

File: src/main/java/com/example/demo/Order_item/Order_item.java (hunk 7 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Order_item/Order_item.java]: done

File: src/main/java/com/example/demo/Product/Product.java (hunk 8 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Product/Product.java]: done

File: src/main/java/com/example/demo/DemoApplication.java (hunk 3 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/DemoApplication.java]: done

File: src/main/java/com/example/demo/CardData/CardDataService.java (hunk 4 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/CardData/CardDataService.java]: done

File: src/main/java/com/example/demo/CardData/CardData.java (hunk 6 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/CardData/CardData.java]: done

File: src/main/java/com/example/demo/CardData/CardData.java (hunk 1 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/CardData/CardData.java]: done

File: src/main/java/com/example/demo/Category/Category.java (hunk 2 above)
Comment (KacFlor) [reply src/main/java/com/example/demo/Category/Category.java]: done

File: pom.xml (hunk 9 above)
Comment (KacFlor) [reply pom.xml:1]: done

File: src/main/java/com/example/shopSpring/model/CardData.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.Table;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table(name = "cardData")
+@Getter
+@Setter
+public class CardData {
```
Comment (Ficzer) [inline src/main/java/com/example/shopSpring/model/CardData.java]: Every model class should extend BaseEntity. Look: https://github.com/spring-projects/spring-petclinic/blob/main/src/main/java/org/springframework/samples/petclinic/model/Person.java

File: src/main/java/com/example/shopSpring/model/CardData.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Id
+    private Integer id;
+
+    private String cardNum;
```
Comment (Ficzer) [inline src/main/java/com/example/shopSpring/model/CardData.java]: Every data field should have @Column annotation right? I dont remember it but is is present in this example: https://github.com/spring-projects/spring-petclinic/blob/main/src/main/java/org/springframework/samples/petclinic/model/Person.java

File: src/main/java/com/example/shopSpring/model/CardData.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.Setter;
+
+@Entity
+@Table(name = "cardData")
+@Getter
+@Setter
+public class CardData {
+
+    @Id
+    private Integer id;
```
Comment (Ficzer) [inline src/main/java/com/example/shopSpring/model/CardData.java]: You dont need id when you extend BaseEntity

File: src/main/java/com/example/shopSpring/model/Customer.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+
+    private Number phoneNumber;
+
+    @OneToMany
+    @JoinColumn
+    private List<Order> orders;
+
+    @OneToMany
+    @JoinColumn
+    private List<CardData> cardDatas;
```
Comment (Ficzer) [inline src/main/java/com/example/shopSpring/model/Customer.java]: Where are reviews, user, cart, shipments?

File: src/main/java/com/example/shopSpring/model/CardData.java (hunk 10 above)
Comment (KacFlor) [reply src/main/java/com/example/shopSpring/model/CardData.java]: done

File: src/main/java/com/example/shopSpring/model/CardData.java (hunk 11 above)
Comment (KacFlor) [reply src/main/java/com/example/shopSpring/model/CardData.java]: done

File: src/main/java/com/example/shopSpring/model/CardData.java (hunk 12 above)
Comment (KacFlor) [reply src/main/java/com/example/shopSpring/model/CardData.java]: done

File: src/main/java/com/example/shopSpring/model/Customer.java (hunk 13 above)
Comment (KacFlor) [reply src/main/java/com/example/shopSpring/model/Customer.java]: done

File: src/main/java/com/KacFlor/ShopSpring/model/Payment.java [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    @Column(name = "amount")
+    private Double amount;
+
+    @OneToOne(mappedBy = "payment", fetch = FetchType.EAGER)
+    private Shipment shipment;
+
+    public Payment() {
+    }
+
+    public Payment(LocalDate paymentDate, String paymentMet, Double amount) {
```
Comment (Ficzer) [inline src/main/java/com/KacFlor/ShopSpring/model/Payment.java:32]: We need to set id of the BaseEntity super class

Comment (KacFlor) [reply src/main/java/com/KacFlor/ShopSpring/model/Payment.java:32]: done

**Review by Ficzer — APPROVED** [review]

**Pull request merged**